### ISSUE TITLE
The way to use glGetError to debug

### ISSUE THREAD

**Issue by liyi14** [glumpy/glumpy#120]
Hi. I met the error code 1281 when executing app.run, while I think the error is more likely at app.Window. 

According to the [link](https://stackoverflow.com/questions/30124343/opengl-error-1281-when-trying-to-gluseprogram) , I can use glGetError to locate the error accurately. 

So my question is that is it possible for me to have similar code in glumpy?

Thank you.

**Comment by rougier** [comment]
Yes, the error is normally retrieved (and even parsed) to display a useful message. What is the code you're running and what is the output ?

**Comment by liyi14** [comment]
The code is a common render code, which is nearly the same as the brain example.
I wrap it into a class and call this class twice.
I can run it the first time or only call it one time, but if I try to build two renderer(two app.Window), I will get following error:
GLError: GLError(
        err = 1281,
        description = 'invalid value',
        baseOperation = glUseProgram,
        cArguments = (1L,)
)

I was just wondering how to make sure whether it failed when I set app.Window the second time or failed when drawing.

**Comment by rougier** [comment]
Did you check for the examples with two windows to see if it works ?

**Comment by liyi14** [comment]
Yes, it works. I'm working glumpy with mxnet, a deep earning framework, and not sure whether there will be any conflict between them.

**Comment by rougier** [comment]
I don't see any obvious  reason why it would fail.

**Comment by liyi14** [comment]
I am also puzzled about it.
By the way, do you have any idea what will lead to the error "ValueError: Cannot create program object".
If I create app.Window in the layer init function and do app.run(framecount=0) in the layer forward function, glumpy will report me this error.

**Comment by rougier** [comment]
Can you start from the two windows example in the examples directory and add some mxnet calls to check when it fails ?

**Comment by liyi14** [comment]
I just solved the problem by the call setup two times, one in layer init and the other in the first time forward, although I don't know why.
In summary:
setup in init: ValueError: "Cannot create program object"
setup in first forward: only one instance is ok, while will have "err = 1281" if I want two instances
setup in init and replace it in first forward: can work.

**Comment by rougier** [comment]
Can you post a minimal script ?
